src/lib.rs

```diff
@@ -62,30 +63,46 @@ impl Decoder {
             }
 
             Decoder {
-                dec: dec.assume_init(),
+                dec: ptr::NonNull::new(dec.assume_init()).unwrap(),
+                pending_data: None,
             }
         }
     }
 
-    pub fn flush(&self) {
+    pub fn flush(&mut self) {
         unsafe {
-            dav1d_flush(self.dec);
+            dav1d_flush(self.dec.as_ptr());
+            if let Some(mut pending_data) = self.pending_data.take() {
+                dav1d_data_unref(&mut pending_data);
+            }
         }
     }
 
-    pub fn send_data<T: AsRef<[u8]>>(
+    pub fn send_data<T: AsRef<[u8]> + Send + 'static>(
         &mut self,
         buf: T,
         offset: Option<i64>,
         timestamp: Option<i64>,
         duration: Option<i64>,
     ) -> Result<(), Error> {
-        let buf = buf.as_ref();
-        let len = buf.len();
+        assert!(
```

Comment:
Wound be nice to not have this footgun, probably an error would be better since it easily recoverable.

Reply:
The problem is that ownership of `buf` was passed to this function, so we'd need to return it back to the caller together with the error or it would be lost. That's why I decided to go with a panic here because it's a programmer error and only can happen if the API is used wrong.

Reply:
Let's model it after [Sender::send()](https://docs.rs/crossbeam-channel/latest/crossbeam_channel/struct.Sender.html#method.send).

Reply:
As discussed on IRC, that's unfortunately not possible because this assertion is the only case where we can safely return the input data. And this case can only happen if the caller uses the API wrong, for which I wouldn't complicate the API.

Any caller would `unwrap()` it most likely anyway as... a caller gracefully handling API-misuse seems very unlikely.